### PR TITLE
make Python optional

### PR DESCRIPTION
**Pull request by casperdcl**
- related #280 <- #266 <- #231

### CODE REVIEW

**Review by 0x2b3bfa0 — COMMENTED** [review]
Should we use `command -v $tool 2>&1 > /dev/null` to be POSIX friendly?

**Comment by casperdcl** [comment]
I don't follow but happy to accept commit suggestions...

File: task/common/machine/script.go [hunk 1]
```diff
@@ -60,12 +60,24 @@ sudo mv terraform-provider-iterative* /usr/bin/tpi
...
+}
+
+if [[ -z "$(which rclone)" ]]; then
+  curl --remote-name https://downloads.rclone.org/rclone-current-linux-amd64.zip
+  extract_here rclone-current-linux-amd64.zip
+  sudo cp rclone-*-linux-amd64/rclone /usr/bin
+  sudo chmod u=rwx,g=rx,o=rx /usr/bin/rclone
+  sudo chown root:root /usr/bin/rclone
+  rm --recursive rclone-*-linux-amd64*
+fi
```
Comment (0x2b3bfa0) [inline task/common/machine/script.go:80]:
```suggestion
extract_here(){
  if command -v unzip 2>&1 > /dev/null; then
    unzip "$1"
  elif command -v python3 2>&1 > /dev/null; then
    python3 -m zipfile -e "$1" .
  else
    python -m zipfile -e "$1" .
  fi
}

if ! command -v rclone 2>&1 > /dev/null; then
  curl --remote-name https://downloads.rclone.org/rclone-current-linux-amd64.zip
  extract_here rclone-current-linux-amd64.zip
  sudo cp rclone-*-linux-amd64/rclone /usr/bin
  sudo chmod u=rwx,g=rx,o=rx /usr/bin/rclone
  sudo chown root:root /usr/bin/rclone
  rm --recursive rclone-*-linux-amd64*
fi
```

See also: [`command`](https://pubs.opengroup.org/onlinepubs/007904975/utilities/command.html)

**Comment by 0x2b3bfa0** [comment]
> _I don't follow but happy to accept commit suggestions..._

Sorry! My previous comment was too obscure.

File: task/common/machine/script.go (hunk 1 above)
Comment (casperdcl) [reply task/common/machine/script.go:80]: sure, if this is more POSIXy...

Comment (0x2b3bfa0) [reply task/common/machine/script.go:80]: I don't see you very convinced. Let me add an example:

### Built–in `which`

```console
$ which which
which: shell built-in command
$ result="$(which unobtanium)"
$ printf "result: %s\n" "$result"
result: unobtanium not found
$ if test -n "$result"; then echo exists; else echo does not exist; fi
exists
```

### External `which`

```console
$ which /usr/bin/which
/usr/bin/which
$ result="$(/usr/bin/which unobtanium)"
$ printf "result: %s\n" "$result"
result: 
$ if test -n "$result"; then echo exists; else echo does not exist; fi
does not exist
```

### Portable `command`

```console
$ which command
command: shell built-in command
$ if command -v unobtanium 2>&1 > /dev/null; then echo exists; else echo does not exist; fi
does not exist
```

Running macOS 10.15.7 and `zsh`

Comment (casperdcl) [reply task/common/machine/script.go:80]: :no-mac:

Comment (0x2b3bfa0) [reply task/common/machine/script.go]: Did you mean `:no-zsh:` instead?

**Comment by 0x2b3bfa0** [comment]
Passes GCP, _ergo_ it works.

AWS is broken because I need to update the MFA sandbox credentials with the CI sandbox ones; AZ is broken because of #264 issues I need to investigate; K8S is broken because of #288.

**Pull request merged**